client/my-sites/customer-home/controller.jsx

```diff
@@ -37,9 +41,18 @@ export async function maybeRedirect( context, next ) {
 	}
 
 	const siteId = getSelectedSiteId( state );
-	const { launchpad_screen, site_intent } = await fetchLaunchpad( slug );
+	const site = getSelectedSite( state );
+	const isSiteLaunched = site?.launch_status === 'launched' || false;
+	const { launchpad_screen, checklist_statuses, site_intent } = await fetchLaunchpad( slug );
+
+	function areLaunchpadTasksCompleted() {
+		if ( 'newsletter' === site_intent ) {
+			return Boolean( checklist_statuses?.first_post_published );
+		}
+		return isSiteLaunched || Boolean( checklist_statuses?.site_launched );
+	}
```

Comment:
Worth noting the two areLaunchpadTasksComplete() method used in the two place here are almost, but not quite, identical. May be worth trying to consolidate those to a single method that is imported and used in both places.

Reply:
1. Why can't they be identical?  This version seems to check the `site_intent` while the other seems to check the flow. We should ensure the flow matches the site intent - the site intent is the main source of truth. The flow is just info from the URL iirc? Right now we can load any flow via the URL, but perhaps those should always be redirecting to the launchpad checklist corresponding to actual `site_intent`.

2. We are still doing a blanket generalization that all intents other than newsletter mark completion via site launch. this might be ok for now, but will cause bugs and need to be updated as soon as anyone adds any other checklist that doesn't - we should probably be more explicit about checking the intent/flow before raising conditional logic based on launch status.

Reply:
I've consolidated to a single method for these - so they're now identical. I've also updated the logic for checking tasks to be more robust, but I'm still not checking site_intent vs flow. I've added a separate comment/thought on that [here](https://github.com/Automattic/wp-calypso/pull/74537#discussion_r1147893482).